### ISSUE TITLE
Load alarm warn/crit values way too high

### ISSUE THREAD

**Issue by kaiomatico** [netdata/netdata#10966]
##### Bug report summary

Commit https://github.com/ilyam8/netdata/commit/2adb71f2af42d6f0e78e13eade8cc262f0d42e2b broke the load alarm warn/crit values because they are now too high by the factor of * 10 to 100

See this example screenshot after running the command `stress-ng --cpu 32 --timeout 120s` on a 8 core CPU which clearly produces an amount of load worth alarming:
![grafik](https://user-images.githubusercontent.com/8037196/114541165-a38d5080-9c56-11eb-97ce-a33160dcb507.png)

##### OS / Environment

```
Linux 9310 5.11.0-13-generic #14-Ubuntu SMP Fri Mar 19 16:55:27 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=21.04
/etc/lsb-release:DISTRIB_CODENAME=hirsute
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu Hirsute Hippo (development branch)"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="21.04 (Hirsute Hippo)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu Hirsute Hippo (development branch)"
/etc/os-release:VERSION_ID="21.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=hirsute
/etc/os-release:UBUNTU_CODENAME=hirsute
```

##### Netdata version

```
Version: netdata v1.30.1
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-option-checking' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--disable-dependency-tracking' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' '--with-bundled-lws=externaldeps/libwebsockets' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Features:
    dbengine:                YES
    Native HTTPS:            YES
    Netdata Cloud:           YES 
    Cloud Implementation:    Legacy
    TLS Host Verification:   YES
Libraries:
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    LWS:                     YES static v3.2.2
    mosquitto:               YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES

```

##### Installation method

Netdata installed via apt packagecloud package 

##### Component Name

Load Alarm

##### Steps To Reproduce

1. stress-ng --cpu 32 --timeout 120s
2. Observe a load around 30 -> which makes the system barely usable, yet is still way too low as avalue to trigger the warn/crit values of the alarm

##### Expected behavior

Warn with a load of around 2

**Comment by ilyam8** [comment]
Hi @kaiomatico

Let's say
- `lookup: max -1m unaligned of load15`: 10
- `$load_trigger` (==`$load_cpu_number`): 8.

---

### tag v1.29.0

https://github.com/netdata/netdata/blob/9580ed4d13125adac01483bac14a18c952afd13e/health/health.d/load.conf#L23-L27

`warn`: 10 > 2 * 8 => 10 > 16 (% of limit: 10 / 16.0 * 100 => 62.5%)

### tag v1.31.0 (current)

https://github.com/netdata/netdata/blob/d5ea3cc5cf7256109c4c76148b5ba0d29328b380/health/health.d/load.conf#L24-L27

`warn`: 10 * 100 / 8 > 200 => 125.0 > 200 (% of limit: 125 / 200.0 * 100 => 62.5%)

---

As you see nothing changed in terms of calculations and trigger conditions/limits.

We use this interpretation of load average

> If the value is higher than your CPU count, then you might have a performance problem (it depends).

So if you have 8 cpu/cores then 100% is 8. That commit changed calculation logic to simplify it: trigger if your load average is 175%/200%/etc of your desired limit.

**Comment by ilyam8** [comment]
load average doesn't always reflect a system _usable_ status, e.g 8 could be ok for one system and not ok for other system.

See https://github.com/torvalds/linux/blob/7f75285ca572eaabc028cf78c6ab5473d0d160be/kernel/sched/loadavg.c#L1-L8

It measures the number of threads that aren't completely idle, i am not sure it's purpose to indicate that your system _barely usable_ or not.

> stress-ng --cpu 32 --timeout 120s

In that case i expect cpu usage alarm to get active.

**Comment by kaiomatico** [comment]
Oooooh I see, so the load is not divided by the core count so it is not normalized to 1, but instead the percentage is now multiplied by the core count instead, am I right? If so then feel free to close this issue and thank you very much for clarifying :+1:

**Comment by ilyam8** [comment]
- load15: [trigger when >= 200%](https://github.com/netdata/netdata/blob/cec48d37ec051a115bd477b94bcf99def89723e7/health/health.d/load.conf#L27)
- load5: [trigger when >= 400%](https://github.com/netdata/netdata/blob/d5ea3cc5cf7256109c4c76148b5ba0d29328b380/health/health.d/load.conf#L39)
- load1: [trigger when >= 800%](https://github.com/netdata/netdata/blob/d5ea3cc5cf7256109c4c76148b5ba0d29328b380/health/health.d/load.conf#L51)

So we don't really want to trigger load1 and load5 if they are not super high, because it could be a relatively short spike.

To be honest i am not happy with current trigger limits, we discussed it with @vlvkobal and couldn't agree on changes and decided to leave them unchanged.

**Comment by kaiomatico** [comment]
It is now clear to me because I missed that $this is being multiplied by 100. Thank you for clarifying :-)  I will close this issue now, have a nice day! :)

**Comment by ilyam8** [comment]
>  I was just confused that $this was so low compared to the trigger

That is understandable. 

I was considering to use [calc](https://learn.netdata.cloud/docs/agent/health/reference#alarm-line-calc), like

```yaml
   alarm: load_average_15
  lookup: max -1m unaligned of load15
   calc:  $this * 100 / $load_cpu_number
   units: %
   every: 1m
    warn: $this > (($status >= $WARNING) ? 175 : 200)
```

but in that case the value would be % and users would be wondering _how come my load average is %_

**Comment by kaiomatico** [comment]
So why not something like:
```
   alarm: load_average_15
  lookup: max -1m unaligned of load15
    calc: $this / $load_cpu_number
   every: 1m
    warn: $this > (($status >= $WARNING) ? 1.75 : 2)
```
?

**Comment by Ferroin** [comment]
We would want to normalize for the check but report actual values for any alerts if we were to take that approach. I don’t know about you, but I’d think it rather strange if my Ryzen 9 3950 home server were reporting a warning for the load average being 8.0 (what it would look like if we normalized to a single core equivalent) given that it’s a 16-core CPU with 2-way SMT, especially when I then looked at the system itself and saw completely different numbers.

Ideally, we would have a way to handle things like that, but we currently don’t really. If we did it would probably be the best approach here though.
